src/vcpkg/binarycaching.cpp

```diff
@@ -1455,6 +1496,34 @@ namespace vcpkg
         return results;
     }
 
+    void BinaryCache::push_thread_main()
+    {
+        while (true)
+        {
+            std::unique_lock<std::mutex> lock(actions_to_push_mutex);
+            actions_to_push_notifier.wait(lock, [this]() { return !actions_to_push.empty() || end_push_thread; });
+            if (actions_to_push.empty())
+            {
+                if (end_push_thread) break;
+                continue;
+            }
+
+            auto entry = std::move(actions_to_push.front());
+            actions_to_push.pop();
+            lock.unlock(); // we don't touch actions_to_push anymore
```

Comment:
Use a scope to release the lock?

Reply:
But the entry is not copyable and I would have to declare it outside of the scope and assign it inside of the scope